client/src/components/categorical/category.js

```diff
@@ -200,6 +200,9 @@ class Category extends React.Component {
     /* allow empty string */
     if (name === "") return false;
 
+    /* allow any term in the ontology */
+    if (this.props.ontology.terms.includes(name)) return false;
```

Comment:
if the ontology is large, this is going to be slow/laggy (remember, it gets evaluated upon each keystroke event).

I suggest changing the component to have a derived property which is a Set object containing the ontology terms, and testing for the input value presence in the set.

Reply:
Second thought - you will need this same query in categorical.js.  So you might as well create a single shared Set object in the reducer, and store in the main app state.  That way the two components do not create duplicate objects.